cmd/agent/app/integrations.go

```diff
@@ -40,6 +42,28 @@ var (
 	tufConfig    string
 )
 
+type versionStruct struct {
+	major int
+	minor int
+	fix   int
+}
+
+func newVersion(version string) (versionStruct, error) {
```

Comment:
nit on Go idioms here: a `new`-prefixed constructor should return a pointer to the type. You could then work with pointers in the rest of the code, and make all functions that return the struct directly return a pointer instead.

Also, another nit: it may be just a personal preference, but I would call this differently from `new[...]`, since we're actually parsing the version that's passed, not just allocating a new versionStruct, if that makes sense. So I'd name it `func parseVersion(version string) (*versionStruct, error)` instead, or something similar.